grakn-core/src/main/java/ai/grakn/concept/Relation.java

```diff
@@ -66,9 +67,19 @@
      *
      * @return A list of all the Instances and their Role Types.
      */
+    @Deprecated // TODO: Remove this method
     Map<RoleType, Instance> rolePlayers();
 
     /**
+     * Retrieves a list of every {@link Instance} involved in the {@link Relation}, filtered by {@link RoleType} played.
+     * @param roleTypes used to filter the returned instances only to ones that play any of the role types.
+     *                  If blank, returns all role players.
+     * @return a list of every {@link Instance} involved in the {@link Relation}.
+     */
+    // TODO: Rename this method to `rolePlayers`
+    Collection<Instance> newRolePlayers(RoleType... roleTypes);
```

Comment:
That's true... I added this method to parallel `Instance#relations`, which has the same issues you describe. I'd still keep it, because the alternative `Map<RoleType, Collection<Instance>>` is worse, in my opinion.